cmd/azbrowse/cmd.go

```diff
@@ -98,26 +102,52 @@ func createRootCmd() *cobra.Command {
 	cmd.Flags().StringVar(&tenantID, "tenant-id", "", "(optional) specify the tenant id to get an access token for (see az account list -o json)")
 	cmd.Flags().StringVar(&subscription, "subscription", "", "(optional) specify a subscription to load")
 
-	if err := cmd.RegisterFlagCompletionFunc("subscription", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if err := cmd.RegisterFlagCompletionFunc("subscription", subscriptionAutocompletion); err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+// This allows azbrowse to update the account cache used for autocompletion
+func updateAccountCache() {
+	out, err := exec.Command("az", "account", "list", "--query", "[].[name, id] | [] | sort(@)", "--output", "tsv").Output()
+	if err != nil {
+		return
+	}
+	err = storage.PutCacheForTTL(keyForAccountCache, string(out))
+	if err != nil {
+		panic("Failed to save account list to cache")
+	}
+}
+
+// Provide support for autocompleting subscriptions
+func subscriptionAutocompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	var accountList []byte
 
+	validCache, value, err := storage.GetCacheWithTTL(keyForAccountCache, time.Hour*6)
+	if !validCache || err != nil {
 		out, err := exec.Command("az", "account", "list", "--query", "[].[name, id] | [] | sort(@)", "--output", "tsv").Output()
```

Comment:
Does it make sense to reuse the `updateAccountCache` function here?

Reply:
Yes, good catch - I'll fix that up.

Reply:
Done, just giving it a quick test